Projects/UOContent/Mobiles/AI/BaseAI.cs

```diff
@@ -993,106 +899,131 @@ public virtual bool DoActionCombat()
+    {
+        return combatant != null && !combatant.Deleted && combatant.Map == m_Mobile.Map
+            && combatant.Alive && !combatant.IsDeadBondedPet;
+    }
 
     public virtual bool DoActionGuard()
     {
-        if (Core.AOS && CheckHerding())
-        {
-            if (m_Mobile.Debug)
-            {
-                m_Mobile.DebugSay("Praise the shepherd!");
-            }
-        }
-        else if (Core.TickCount - m_NextStopGuard < 0)
+        if (Core.TickCount - m_NextStopGuard < 0)
         {
             if (m_Mobile.Debug)
             {
-                m_Mobile.DebugSay("I am on guard");
+                m_Mobile.DebugSay("I am still on guard.");
             }
-            // m_Mobile.Turn( Utility.Random(0, 2) - 1 );
+
+            m_Mobile.Turn(Utility.Random(0, 2) - 1); // added for immersion
         }
         else
         {
             if (m_Mobile.Debug)
             {
-                m_Mobile.DebugSay("I stopped being on guard");
+                m_Mobile.DebugSay("I stopped being on guard.");
             }
 
-            Action = ActionType.Wander;
+            if (Action != ActionType.Wander)
+            {
+                Action = ActionType.Wander;
+            }
         }
-
+    
         return true;
     }
 
     public virtual bool DoActionFlee()
     {
         var from = m_Mobile.FocusMob;
-
-        if (from?.Deleted != false || from.Map != m_Mobile.Map)
+    
+        if (!IsValidFocusMob(from))
         {
             if (m_Mobile.Debug)
             {
-                m_Mobile.DebugSay("I have lost him");
+                m_Mobile.DebugSay("Focused target is missing.");
             }
 
-            Action = ActionType.Guard;
+            if (Action != ActionType.Guard)
+            {
+                Action = ActionType.Guard;
+            }
             return true;
         }
-
-        if (WalkMobileRange(from, 1, true, m_Mobile.RangePerception * 2, m_Mobile.RangePerception * 3))
+    
+        if (WalkMobileRange(from, 1, true, m_Mobile.RangePerception * 2, m_Mobile.RangePerception * 1))
         {
             if (m_Mobile.Debug)
             {
-                m_Mobile.DebugSay("I have fled");
+                m_Mobile.DebugSay("I am fleeing.");
             }
 
-            Action = ActionType.Guard;
+            if (Action != ActionType.Guard)
+            {
+                Action = ActionType.Guard;
+            }
             return true;
         }
-
+        
         if (m_Mobile.Debug)
         {
             m_Mobile.DebugSay("I am fleeing!");
         }
 
         return true;
     }
+    
+    private bool IsValidFocusMob(Mobile focusMob)
+    {
+        return focusMob != null && !focusMob.Deleted && focusMob.Map == m_Mobile.Map;
+    }
 
-    public virtual bool DoActionInteract() => true;
+    private void DebugSay(string message)
+    {
+        if (m_Mobile.Debug)
+        {
+            m_Mobile.DebugSay(message);
+        }
+    }
```

Comment:
We need a better way of doing this, and this makes it worse because we don't want to do the interpolation into a string unless debug is enabled. I'll look into a better way.